### PR TITLE
[PLU-102] feat: rich text editor input field

### PR DESCRIPTION
**Pull request by stanleynguyen**
## Problem

We have input fields that take HTML content, however, our current inputs are only able to do plain text. Hence, we have a two-fold problem:
- We have to massage the plaintext content produced by the plaintext editor, and if users decide to input raw HTML this content will be affected by the massaging that we do on the plaintext content as well
- Users have no easy way to preview the raw HTML content that they're putting in

## Solution

Implement a new type of input `rich-text` that uses tiptap

## Tests

All tests from #273 #274 #277 

## Deploy Notes

- [ ] Send emails to notify all users that might be affected if they decide to update the affected fields from their existing pipes

### CODE REVIEW

**Review by ogp-weeloong — APPROVED** [review]
LGTM

If possible, would be good to fix the variables popper obscuring the tiptap menu if window height is too small

**Review by ogp-weeloong — DISMISSED** [review]
wait wait I take back what I said, old pipes break even without editing because we are not translating `\n` to `<br>` anymore.

https://github.com/opengovsg/plumber/pull/277#discussion_r1396999735 

#278 needs to be merged in before this entire feature branch can be merged into `develop`.

**Comment by stanleynguyen** [comment]
> wait wait I take back what I said, old pipes break even without editing because we are not translating `\n` to `<br>` anymore.
> 
> [#277 (comment)](https://github.com/opengovsg/plumber/pull/277#discussion_r1396999735)
> 
> #278 needs to be merged in before this entire feature branch can be merged into `develop`.

Actually I wanna merge #278 after we have merged and release this, to limit the impact to just the email body field, so I've brought up the changes needed from #277

File: packages/frontend/src/components/RichTextEditor/ResizableImageExtension.tsx
Comment (m0nggh) [inline packages/frontend/src/components/RichTextEditor/ResizableImageExtension.tsx]: need add missing dependency `updateAttributes`

File: packages/frontend/src/components/RichTextEditor/SuggestionPopper.tsx
Comment (m0nggh) [inline packages/frontend/src/components/RichTextEditor/SuggestionPopper.tsx:12]: is this MUI gonna be changed in a subsequent PR?

File: packages/frontend/src/components/RichTextEditor/ResizableImageExtension.tsx
Comment (m0nggh) [inline packages/frontend/src/components/RichTextEditor/ResizableImageExtension.tsx]: nit: can remove the `React.`

File: packages/frontend/src/components/RichTextEditor/SuggestionPopper.tsx
Comment (stanleynguyen) [reply packages/frontend/src/components/RichTextEditor/SuggestionPopper.tsx:12]: No this one is intended. Approved by Stace jiejie alr - "As long as the component looks like ODS, it's ok"

File: packages/frontend/src/components/RichTextEditor/ResizableImageExtension.tsx
Comment (stanleynguyen) [reply packages/frontend/src/components/RichTextEditor/ResizableImageExtension.tsx]: This one cannot:
- it's to differentiate btwn React MouseEvent vs native MouseEvent
- also linting will behave weirdly if we import MouseEvent interface directly from 'react' 🤔

**Review by m0nggh — APPROVED** [review]
okay other than the image not appearing on the editor for some links, think its good to go, lgtm

File: packages/frontend/src/components/NewsDrawer/NewsItemList.ts
Comment (m0nggh) [inline packages/frontend/src/components/NewsDrawer/NewsItemList.ts]: eh small nit: `brand new`

File: packages/frontend/src/components/NewsDrawer/NewsItemList.ts
Comment (m0nggh) [inline packages/frontend/src/components/NewsDrawer/NewsItemList.ts]: and also `backward compatibility`, thanksss

**Pull request merged**
